### PR TITLE
Support for UTF-8 content. Hebrew import now preserve encoding

### PR DESCRIPTION
**Pull request by eliagrady**
I don't know why, but 'CharStreams.copy(reader, writer)' would kill the UTF-8 hebrew encoding.
What I've done is to detect UTF-8 and bypass the copy method with raw byte copy.
Hebrew characters are now preserved via 'import'.

### CODE REVIEW

File: src/main/be/wimsymons/intellij/polopolyimport/PPImporter.java [hunk 1]
```diff
@@ -291,4 +295,16 @@ private Reader wrapWithReplacements(InputStream in, Charset charset) {
 			return ReplacementsReaderBuilder.with(new InputStreamReader(in, charset), replacements);
 		}
 	}
+
+  private void copyAndFlush(Reader reader, OutputStream writeTo) throws IOException {
+    InputStream stream = new ReaderInputStream(reader, "UTF-8");
```
Comment (wimsymons) [inline src/main/be/wimsymons/intellij/polopolyimport/PPImporter.java]: Why not use https://docs.oracle.com/javase/7/docs/api/java/io/InputStreamReader.html?

File: src/main/be/wimsymons/intellij/polopolyimport/PPImporter.java [hunk 2]
```diff
@@ -207,40 +204,47 @@ private void addToJar(JarOutputStream jarOS, VirtualFile file) throws IOExceptio
...
+      httpConnection.setDoInput(true);
+      httpConnection.setDoOutput(true);
+      httpConnection.setUseCaches(false);
+      httpConnection.setRequestMethod("PUT");
+      httpConnection.setRequestProperty("Content-Type", contentType);
+      httpConnection.setConnectTimeout(2000);
+      httpConnection.setReadTimeout(60000);
+      httpConnection.connect();
+
+      if(contentType.contains("UTF-8")) {
```
Comment (wimsymons) [inline src/main/be/wimsymons/intellij/polopolyimport/PPImporter.java]: Shouldn't you do case-insensitive matching to be sure?

File: src/main/be/wimsymons/intellij/polopolyimport/PPImporter.java [hunk 3]
```diff
@@ -110,8 +106,9 @@ private void doImportSingleFile(VirtualFile virtualFile) {
 			InputStream dataIS;
 			try {
 				dataIS = virtualFile.getInputStream();
-				String contentType = "text/xml;charset=" + virtualFile.getCharset();
+				String contentType = "text/xml&charset=" + virtualFile.getCharset();
```
Comment (wimsymons) [inline src/main/be/wimsymons/intellij/polopolyimport/PPImporter.java]: The '&' is wrong. Check https://www.w3.org/Protocols/rfc1341/4_Content-Type.html. It must be a ';'. Maybe a typo?

**Review by wimsymons — CHANGES_REQUESTED** [review]
Some things don't seem correct. Can you check my comments and fix them?

**Comment by wimsymons** [comment]
Other question for you Elia. I no longer use Polopoly at work as a CMS so supporting this plugin is no longer of my interest. You be willing to become the new owner of the code?

File: src/main/be/wimsymons/intellij/polopolyimport/PPImporter.java (hunk 1 above)
Comment (eliagrady) [inline src/main/be/wimsymons/intellij/polopolyimport/PPImporter.java]: The one you've suggested extended Reader.
The one I'm using extends InputStream.
Basically, I've used it in order to get access to the InputStream, that was previously wrapped with a Reader, while not changing the API to pass it down.
Both support the 'read(byte[] b, int off, int len)' call, but the ReaderInputStream also supports 'read(byte[] b)' which InputStreamReader does not.

File: src/main/be/wimsymons/intellij/polopolyimport/PPImporter.java (hunk 2 above)
Comment (eliagrady) [inline src/main/be/wimsymons/intellij/polopolyimport/PPImporter.java]: yes, you're right. 
contentType.toLowerCase().contains("utf-8") should be better.

File: src/main/be/wimsymons/intellij/polopolyimport/PPImporter.java (hunk 3 above)
Comment (eliagrady) [inline src/main/be/wimsymons/intellij/polopolyimport/PPImporter.java]: Not a typo.
Some history: the eclipse plugin was working just fine and I was trying to imitate its behavior.
As you can see at line 20 here:
https://github.com/jwiklund/pp-eclipse/blob/master/src/pp/eclipse/post/ContentImporter.java
They used &.
I should test it further to see if it's some weird implementation artifact from the ATEX importer internal code, or stick with the W3C spec.

**Comment by eliagrady** [comment]
Regarding code ownership, as long as I work with the Atex system - sure - no problem.

**Pull request merged**